iceoryx2/tests/waitset_tests.rs

```diff
@@ -0,0 +1,388 @@
+                }
+            })
+            .unwrap();
+
+        assert_that!(wait_event, eq WaitEvent::Notification);
+
+        assert_that!(listener_1_triggered, eq true);
+        assert_that!(receiver_1_triggered, eq true);
+    }
+
+    #[test]
+    fn try_wait_does_not_block<S: Service>()
+    where
+        <S::Event as Event>::Listener: SynchronousMultiplexing,
+    {
+        let _watchdog = Watchdog::new();
+        let node = NodeBuilder::new().create::<S>().unwrap();
+        let sut = WaitSetBuilder::new().create::<S>().unwrap();
+
+        let (listener, _) = create_event::<S>(&node);
+        let _guard = sut.attach(&listener);
+
+        let wait_event = sut
+            .try_wait(|_| {
+                test_fail!("only attachments shall trigger");
+            })
+            .unwrap();
+
+        assert_that!(wait_event, eq WaitEvent::Tick);
+    }
+
+    #[test]
+    fn timed_wait_blocks_for_at_least_timeout<S: Service>()
+    where
+        <S::Event as Event>::Listener: SynchronousMultiplexing,
+    {
+        let _watchdog = Watchdog::new();
+        let node = NodeBuilder::new().create::<S>().unwrap();
+        let sut = WaitSetBuilder::new().create::<S>().unwrap();
+
+        let (listener, _) = create_event::<S>(&node);
+        let _guard = sut.attach(&listener);
+
+        let start = Instant::now();
+        let wait_event = sut
+            .timed_wait(
+                |_| {
+                    test_fail!("only attachments shall trigger");
+                },
+                TIMEOUT,
+            )
+            .unwrap();
+
+        assert_that!(wait_event, eq WaitEvent::Tick);
+        assert_that!(start.elapsed(), time_at_least TIMEOUT);
+    }
+
+    #[test]
+    fn blocking_wait_blocks<S: Service + 'static>()
+    where
+        <S::Event as Event>::Listener: SynchronousMultiplexing,
+    {
+        let _watchdog = Watchdog::new();
+        let node = NodeBuilder::new().create::<S>().unwrap();
+        let sut = WaitSetBuilder::new().create::<S>().unwrap();
+
+        let service_name = generate_name();
+        let service = node
+            .service_builder(&service_name)
+            .event()
+            .open_or_create()
+            .unwrap();
+
+        let listener = service.listener_builder().create().unwrap();
+        let _guard = sut.attach(&listener);
+
+        let start = Instant::now();
+        let barrier = Arc::new(Barrier::new(2));
+        let barrier_thread = barrier.clone();
+
+        let t1 = std::thread::spawn(move || {
+            let notifier = service.notifier_builder().create().unwrap();
+            barrier_thread.wait();
+            std::thread::sleep(TIMEOUT);
+            notifier.notify().unwrap();
+        });
+
+        barrier.wait();
+        let wait_event = sut.blocking_wait(|_| {}).unwrap();
+
+        t1.join().unwrap();
+
+        assert_that!(wait_event, eq WaitEvent::Notification);
+        assert_that!(start.elapsed(), time_at_least TIMEOUT);
```

Comment:
```suggestion
        barrier.wait();
        let wait_event = sut.blocking_wait(|_| {}).unwrap();
        
        assert_that!(wait_event, eq WaitEvent::Notification);
        assert_that!(start.elapsed(), time_at_least TIMEOUT);
        
        t1.join().unwrap();
```

The current test checks the sleep in the thread :)